Assets/Scripts/Units/StunnedBehaviour.cs

```diff
@@ -0,0 +1,29 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace RogueApeStudio.Crusader.Units
+{
+    public class StunnedBehaviour
+    {
+        private float _timer;
+        public Enemy Unit { set; private get; }
+
+        public void StartStun()
+        {
+            _timer = 0;
+            Unit.LocalAnimator.SetTrigger("Stunned");
+        }
+
+        public bool StunEnded()
```

Comment:
Any particular reason you're not using `UniTask.WaitForSeconds(time)`

Reply:
No reason, this just works.

Reply:
I'd prefer it if you used `UniTask` in the future, but I'll let it through for now.